src/ingredients/ingredients.controller.ts

```diff
@@ -0,0 +1,101 @@
+import { CreateIngredientDto } from '@/ingredients/dto/create.ingredient.dto';
+import { UpdateIngredientDto } from '@/ingredients/dto/update.ingredient.dto';
+import { IngredientsService } from '@/ingredients/ingredients.service';
+import { IngredientSchema } from '@/schema/ingredients.schema';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  Patch,
+  Post,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@Controller({ path: 'ingredients', version: '1' })
+@ApiTags('ingredients')
+export class IngredientsController {
+  constructor(private readonly ingredientsService: IngredientsService) {}
+
+  // Create a ingredient
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description: 'Ingredient created',
+    type: () => IngredientSchema,
+  })
+  @ApiOperation({
+    summary: 'Create a ingredient',
+  })
+  @Post()
+  @HttpCode(HttpStatus.CREATED)
+  async create(@Body() createIngredientDto: CreateIngredientDto) {
+    return await this.ingredientsService.createIngredient(createIngredientDto);
+  }
```

Comment:
Please add API response documentation for the exception that is thrown from its service

Reply:
Added